helix-view/src/document.rs

```diff
@@ -1108,10 +1107,12 @@ impl Display for FormatterError {
             }
             Self::BrokenStdin => write!(f, "Could not write to formatter stdin"),
             Self::WaitForOutputFailed => write!(f, "Waiting for formatter output failed"),
-            Self::Stderr(output) => write!(f, "Formatter error: {}", output),
             Self::InvalidUtf8Output => write!(f, "Invalid UTF-8 formatter output"),
             Self::DiskReloadError(error) => write!(f, "Error reloading file from disk: {}", error),
-            Self::NonZeroExitStatus => write!(f, "Formatter exited with non zero exit status:"),
+            Self::NonZeroExitStatus(Some(output)) => write!(f, "Formatter error: {}", output),
+            Self::NonZeroExitStatus(None) => {
+                write!(f, "Formatter exited with non zero exit status:")
```

Comment:
I think we should either chop off the trailing `:` here or record and show the status code. IIRC that can be None so we may want `NonZeroExitStatus` to instead be

```rs
NoneZeroExitStatus { code: Option<NonZeroI32>, stderr: Option<String> }
```

Reply:
I don't think showing the status code is really useful, there are very few cases in which it matters since most of the time stderr is used to explain the error. 
Will remove the trailing ':'.